### ISSUE TITLE
Inclusion of Learning rate scheduling 

### ISSUE THREAD

**Issue by GiovanniPioDelvecchio** [GiovanniPioDelvecchio/GCNs_on_text#1]
It is necessary to add a LR scheduling technique in order to achieve a lower bias on both training and validation.
torch already exposes some classes to add LR scheduling to our training recipe:
-  This could be used in order to perform some early tests, it consists in the step lr scheduling: https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#torch.optim.lr_scheduler.StepLR
- This could be used to implement the cosine lr scheduling often used in CV: https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LambdaLR.html#torch.optim.lr_scheduler.LambdaLR
